src/client/components/ProductDetails/ProductDetails.component.js

```diff
@@ -1,82 +1,106 @@
-import React from 'react'
+import React from 'react';
 import PropTypes from 'prop-types';
-import NumberInput from '../NumberInput/NumberInput.component'
-import ButtonComponent from '../Button/Button.component'
-import './ProductDetails.style.css'
+import NumberInput from '../NumberInput/NumberInput.component';
+import ButtonComponent from '../Button/Button.component';
+import './ProductDetails.style.css';
 import Heart from './Heart';
 
+export const ProductDetails = ({
+  imgSource,
+  ProductName,
+  RemainingUnit,
+  Price,
+  productColor,
+  productSize,
+  onClick,
+  isFavorite,
+  imageAlt,
+}) => {
+  const [color, setColor] = React.useState('');
+  const [size, setSize] = React.useState('');
+  const [quantity, setQuantity] = React.useState('');
+  const [checked, setChecked] = React.useState(isFavorite);
 
-export const ProductDetails = (
-    {
-        imgSource,
-        ProductName,
-        RemainingUnit,
-        Price,
-        productColor,
-        productSize,
-        onClick,
-        isFavorite,
-        imageAlt,
-    }
-) => {
-    const checkFavorite = () => {
-        console.log('checked') // This function need to be change database for add or remove from favorite.
-    }
+  const checkFavorite = () => {
+    // This function need to be change database for add or remove from favorite.
+    setChecked(!checked);
+  };
+  const getQuantity = (val) => {
+    setQuantity(val);
+  };
 
-    return <div id="product-details">
-        <div className='product-image'>
-            <img src={imgSource} alt={imageAlt} />
+  return {
+    color,
+    size,
+    quantity,
```

Comment:
You should not export them. These will already be posted in the component for add shopping card

Reply:
according to my understanding , product page will pass these values "add to cart" page. so product detail component must export these parameter.
please look at Julia NumberInput component.